### PR TITLE
[AZ-577 & AZ-576] Feature/rust toolchain

### PR DESCRIPTION
**Pull request by fbielejec**
- closes [AZ-577](https://cardinal-cryptography.atlassian.net/browse/AZ-577) 
- closes [AZ-576](https://cardinal-cryptography.atlassian.net/browse/AZ-576)

### CODE REVIEW

File: .github/workflows/unit_tests.yml [hunk 1]
```diff
@@ -12,26 +12,20 @@ jobs:
...
         uses: actions-rs/toolchain@v1
-        with:
-          profile: minimal
-          toolchain: ${{ env.RUST_TOOLCHAIN_VERSION }}
-          override: true
-          components: rustfmt, clippy
 
-      - name: Install WASM target
-        run: rustup target add wasm32-unknown-unknown --toolchain "$RUST_TOOLCHAIN_VERSION"
+      - name: Install clippy and fmt
```
Comment (h4nsu) [inline .github/workflows/unit_tests.yml:23]: Idea to consider: we could move this and python linting to a seperate workflow. It's a major PITA to find out python linting is off having spent 30 minutes waiting for unit tests to finish

**Review by h4nsu — APPROVED** [review]
LGTM
Two optional suggestions:
- move linting to a separate workflow
- bump Python versions to something up to date, like 3.10

**Review by Yithis — APPROVED** [review]
LGTM

File: .github/workflows/unit_tests.yml (hunk 1 above)
Comment (fbielejec) [reply .github/workflows/unit_tests.yml:23]: make a jira ticket pls?

**Pull request merged**